src/main/java/org/fcrepo/auth/webac/LinkHeaderProvider.java

```diff
@@ -69,15 +73,20 @@
         final ListMultimap<String, String> headers = ArrayListMultimap.create();
 
         LOGGER.debug("Adding WebAC Link Header for Resource: {}", resource);
-
-        nodeService.find(internalSession, resource.getPath()).getTriples(translator, PropertiesRdfContext.class)
-        .filter(t -> model.asStatement(t).getPredicate().hasURI(WEBAC_ACCESS_CONTROL_VALUE))
-        .filter(t -> t.getObject().isURI())
-        .forEachRemaining(t -> {
-                    headers.put("Link", Link.fromUri(uriInfo.getBaseUriBuilder()
-                            .path(translator.convert(model.asStatement(t).getObject().asResource()).getPath())
-                            .toString()).rel("acl").build().toString());
-        });
+        // Get the correct Acl for this resource
+        final Optional<Pair<URI, FedoraResource>> acls = WebACRolesProvider.getEffectiveAcl(resource);
+        if (acls.isPresent()) {
+            // If the Acl is present we probably need to use the internal session to get it's URI
+            nodeService.find(internalSession, acls.get().getRight().getPath())
+                    .getTriples(translator, PropertiesRdfContext.class)
+                    .filter(t -> model.asStatement(t).getPredicate().hasURI(WEBAC_ACCESS_CONTROL_VALUE))
+                    .filter(t -> t.getObject().isURI())
```

Comment:
The only valid values for `WEBAC_ACCESS_CONTROL_VALUE` are URIs like so:

    <> a fedora:Resource ;
        acl:accessControl </rest/acls/acl1> .

So this is not a valid statement:

    <> a fedora:Resource ;
        acl:accessControl "/rest/acls/acl1" .

Since the webac module does not enforce (and has no way to enforce) validity on data input (which is the domain of the fcrepo-kernel-modeshape module), this approach is to filter out any literals that might have been included.

Some alternatives would be to:
  * accept literal values if they refer to repository resource paths (a bad idea IMO)
  * throw a runtime exception when resources are accessed with incorrect acl structure (I would also have concerns about this)
  * include a `WARN`-level log and continue as before